src/http.c

```diff
@@ -126,8 +163,8 @@ http_response_cleanup(struct http_response *r, int fd, int success) {
 
 	/* cleanup response object */
 	for(i = 0; i < r->header_count; ++i) {
-		free(r->headers[i].key);
-		free(r->headers[i].val);
+		if(r->headers[i].copy & HEADER_COPY_KEY) free(r->headers[i].key);
+		if(r->headers[i].copy & HEADER_COPY_VALUE) free(r->headers[i].val);
```

Comment:
hmm these use `&` while the ones above use `==`. It does work with the current values assigned to these enum entries, but there's no `HEADER_COPY_BOTH` for example that would have both bits set. It would be better if the two uses were consistent.